core/store/hashtree/mem/path.go

```diff
@@ -0,0 +1,70 @@
+package mem
+
+import (
+	"math/big"
+
+	"go.dedis.ch/dela/crypto"
+)
+
+// Path is a path along the tree nodes to a key and its value, or none.
+//
+// - implements hashtree.Path
+type Path struct {
+	key []byte
+	// Root is the root of the hash tree. This value is not serialized and
+	// reproduced from the leaf and the interior nodes when deserializing.
+	root      []byte
+	interiors [][]byte
```

Comment:
it should be clear that it's only the hashes of the interior nodes, and not their keys. Which make me wonder, why not storing the keys instead of the hashes? Given a tree, if one gives me the path from a child to the node, I am expecting getting all the interior nodes, not only a list of hashes.

Reply:
The key is not useful and you expect the proof to contain the minimum set of information that will allow you to check the integrity.